bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -205,35 +205,62 @@ public BlackboardArtifact.Type getOrAddArtifactType(String typeName, String disp
 			return typeNameToArtifactTypeMap.get(typeName);
 		}
 
-		Statement s = null;
-		ResultSet rs = null;
 		CaseDbTransaction trans = null;
 		try {
 			trans = caseDb.beginTransaction();
 
 			CaseDbConnection connection = trans.getConnection();
```

Comment:
Why open a transaction when the operation is only a read?

Reply:
I believe the connection for the transaction initially checks if the artifact type previously exists, if not, it identifies the next id to use and then inserts the artifact type.  I think the transaction handles performing the function in an atomic matter.